### PR TITLE
Closing animation progress bar

### PR DESCRIPTION
**Pull request by Timo-Thees**
Added features: 
- Progress bar in the popup that appears after saving and closing
- confetti animation that appears when the set writing goal is reached
- fixed undiscovered bugs form previous version
- implemented some styling along the way

sorry, this update got a bit messy

### CODE REVIEW

File: components/ClosingMessage.js [hunk 1]
```diff
@@ -1,5 +1,8 @@
 import {Overlay, Dialog} from "../styles/PopupStyles";
 import {Button} from "./Button";
+import styled from "styled-components";
+import {keyframes} from "styled-components";
```
Comment (julialoeschel) [inline components/ClosingMessage.js:4]: import styled, {keyframes} from "styled-components";

File: components/ClosingMessage.js [hunk 2]
```diff
@@ -38,17 +41,41 @@ export default function ClosingPopup({
...
+    const progressReportOfToday = dailyProgress.find(
+      progressReport => progressReport.calenderDay === sessionClose.calenderDay
+    );
+    const didIPlanToWriteToday = writingGoals.find(
+      writingGoals => writingGoals.weekday === dailyProgress.weekday
+    );
+    if (didIPlanToWriteToday === undefined) {
+      return "40vw";
+    } else if (progressReportOfToday === undefined) {
+      console.log("progressReportOfToday is undefined");
```
Comment (julialoeschel) [inline components/ClosingMessage.js:68]: this could be an error.

File: components/ClosingMessage.js [hunk 3]
```diff
@@ -38,17 +41,41 @@ export default function ClosingPopup({
...
+      return "40vw";
+    } else {
+      const wordsIWroteToday = progressReportOfToday.wordsIWroteToday * 1;
+      const myGoal = writingGoals[0].writingGoal * 1;
+      const returnThisNumber = (wordsIWroteToday / myGoal) * 40;
```
Comment (julialoeschel) [inline components/ClosingMessage.js:73]: this number is the process, right? I´d prefer to call it what it is ;)

File: components/Timeslot.js [hunk 4]
```diff
@@ -1,11 +1,40 @@
...
   return (
-    <form>
-      <input type="time" name="starting time"></input>
-      <input type="time" name="ending time"></input>
-      <button onClick={() => handleDeleteTimeslot(keyForDeletion, day)}>
+    <form onSubmit={() => handleUpdateTimeslot(keyForDeletion)}>
+      {startingTime !== "" && endingTime !== "" ? (
+        <>
+          <p>
+            In this timeslot you want to write from {startingTime} to{" "}
```
Comment (julialoeschel) [inline components/Timeslot.js:17]:
```suggestion
            In this timeslot you want to write from {startingTime} to
```

**Review by julialoeschel — COMMENTED** [review]
good work!

I have some remarks for you to consider. 
Overall I like your naming, its very descriptive. 

please make sure your next PRs are smaller! 
:octocat:

File: components/ClosingMessage.js (hunk 1 above)
Comment (Timo-Thees) [reply components/ClosingMessage.js:4]: Thanks! Did not know you could import both in the same line this way

File: components/ClosingMessage.js (hunk 2 above)
Comment (Timo-Thees) [reply components/ClosingMessage.js:68]: You are right, made it an error message

File: components/Timeslot.js (hunk 4 above)
Comment (Timo-Thees) [reply components/Timeslot.js:17]: This is something VisualCode puts in on its own. Every time I format it correctly and save its back again.

**Review by actyralf — APPROVED** [review]
Well done!

**Pull request merged**